lib/src/intl/intl_stream.dart

```diff
@@ -5,94 +5,52 @@
 import 'dart:math';
 
 import 'constants.dart' as constants;
-import 'regexp.dart' as regexp;
 
-/// A simple and not particularly general stream class to make parsing
-/// dates from strings simpler. It is general enough to operate on either
-/// lists or strings.
-// TODO(alanknight): With the improvements to the collection libraries
-// since this was written we might be able to get rid of it entirely
-// in favor of e.g. aString.split('') giving us an iterable of one-character
-// strings, or else make the implementation trivial.
+/// An indexed position in a String which can read by specified character
+/// counts, or read digits up to a delimeter.
+// TODO(nbosch): This is too similar to StringIterator for them to both exist.
```

Comment:
If so, then why not merge them? I did read "embrace the yak shave" somewhere in our team values ;)

Reply:
I generally time-box any drive-by refactoring I do. When I started to look at this the scope was creeping up. This change is still an incremental improvement so we may as well ship it.

The commits are more contained this way - merging these two would be a much noisier commit since a lot more usage would need to change.